### ISSUE TITLE
Unable to get the proper regression results

### ISSUE THREAD

**Issue by ZhouCX117** [nzl-thu/MUSDL#3]
Hello, thank you for your wonderful work!
I can get wonderful results for USDL and MUSDL settings. However, the results are terrible when I turn the labels to the initial hard targets(Our-Regression in Table 3). All outputs are around 70. I tried MSE loss and SmoothL1Loss and they all did not work.

The  modification for code is as follows:
1.I modified the output_dim for "USDL" in opts.py, from 101 to 1.
output_dim = {'USDL':1, 'MUSDL': 21}
2.I removed the softmax function on layer3 in evaluator.py
output = self.layer3(x)
3.I regarded the output from the evaluator as the final predicted scores and calculated the loss.
`preds = evaluator(clip_feats.mean(1)).float()
  #preds = compute_score(args.type, probs, data)
  pred_scores.extend([i.item() for i in preds])
  target=torch.tensor(data['final_score']).float().cuda()
  if split == 'train':
      loss = criterion(preds,target)
      optimizer.zero_grad()
      loss.backward()
      optimizer.step()`

If possible,  could you release your code for "Our-Regression" in the paper?

**Comment by nzl-thu** [comment]
Thanks for your interest in our work.

Unfortunately, we did not save the code for regression since it's not related to our proposed method. But for your problem, it seems like you are using the output of evaluator network directly as the predicted score. Since the output is not normalized, it may lead to inferior performance in regression. For the loss function, we used MSELoss.

You can try passing the output through a sigmoid function and then multiplying it with 104.5(the score range of the dataset), and see if it can help:)
